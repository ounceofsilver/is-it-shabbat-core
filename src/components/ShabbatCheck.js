import 'react';
import PropTypes from 'prop-types';
import { DateTime } from 'luxon';

import { isItShabbat } from 'shabbat-logic';

export default function ShabbatCheck(props) {
	const { now, location, children } = props;
	if (!location) {
		return null;
	}
	const { coords: { latitude, longitude } } = location;
	const { period, countDownTo } = isItShabbat(now, latitude, longitude);
	return children(period, countDownTo);
}
ShabbatCheck.propTypes = {
	now: PropTypes.instanceOf(DateTime).isRequired,
	location: PropTypes.shape({
		coords: PropTypes.shape({
			latitude: PropTypes.number,
			longitude: PropTypes.number,
		}),
	}).isRequired,
	children: PropTypes.func,
};

ShabbatCheck.defaultProps = {
	children: () => {},
};
