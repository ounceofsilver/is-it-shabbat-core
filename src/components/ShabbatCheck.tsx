import { DateTime } from 'luxon';
import React from 'react';
import { is, isItShabbat } from 'shabbat-logic';

import { ILocation } from '../state/types';

interface IShabbatCheckProps {
	now: DateTime;
	location: ILocation;
	children: (period: is, countDownTo: DateTime) => JSX.Element;
}

export const ShabbatCheck: React.SFC<IShabbatCheckProps> = ({
	now,
	location,
	children,
}) => {
	if (!location) {
		return null;
	}
	const { coords: { latitude, longitude } } = location;
	const { period, countDownTo } = isItShabbat(now, latitude, longitude);

	return children(period, countDownTo);
};
