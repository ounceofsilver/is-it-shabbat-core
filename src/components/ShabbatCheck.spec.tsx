import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { is } from 'shabbat-logic';

import { local } from '../test';
import { ShabbatCheck } from './ShabbatCheck';

const location = {
	coords: {
		latitude: 42,
		longitude: 73,
	},
};
const now = local();

describe('ShabbatCheck:', () => {
	it('should call child with period and duration', () => {
		const spy = jest.fn().mockReturnValue(<p>Hello</p>);
		const wrapper = renderer.create(
			<ShabbatCheck now={now} location={location}>
				{spy}
			</ShabbatCheck>,
		);

		expect(spy).toHaveBeenCalledTimes(1);
		const [period, duration] = spy.mock.calls[0];
		expect([is.SHABBAT, is.CANDLELIGHTING, is.NOT_SHABBAT]).toEqual(expect.arrayContaining([period]));
		expect(duration.isLuxonDateTime).toBe(true);

		// renders results of inner function
		expect(wrapper.toJSON()).toMatchSnapshot();
	});
});
