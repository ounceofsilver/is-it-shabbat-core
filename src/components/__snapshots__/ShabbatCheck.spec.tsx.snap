// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShabbatCheck: should call child with period and duration 1`] = `
<p>
  Hello
</p>
`;
